eva_assembly_ingestion/job.py

```diff
@@ -0,0 +1,387 @@
+            'clustering_instance': instance,
+            'remapping_config': cfg.config_file,
+            'remapping_required': self.check_remapping_required(source_assembly)
+        }
+
+        for part in ['executable', 'nextflow', 'jar']:
+            remap_cluster_config[part] = cfg[part]
+        with open(remap_cluster_config_file, 'w') as open_file:
+            yaml.safe_dump(remap_cluster_config, open_file)
+        try:
+            command = [
+                cfg['executable']['nextflow'],
+                '-log', remapping_log,
+                'run', nextflow_pipeline,
+                '-params-file', remap_cluster_config_file,
+                '-work-dir', work_dir
+            ]
+            if resume:
+                command.append('-resume')
+            curr_working_dir = os.getcwd()
+            os.chdir(assembly_directory)
+            run_command_with_output('Nextflow remapping process', ' '.join(command))
+        except subprocess.CalledProcessError as e:
+            self.error('Nextflow remapping pipeline failed')
+            self.set_status_failed(source_assembly)
+            raise e
+        finally:
+            os.chdir(curr_working_dir)
+        self.set_status_end(source_assembly)
+        self.count_variants_from_logs(assembly_directory, source_assembly)
+
+    def check_remapping_required(self, source_assembly):
+        return source_assembly != self.target_assembly
+
+    def create_extraction_properties(self, output_file_path, source_assembly):
+        properties = self.properties_generator.get_remapping_extraction_properties(
+            taxonomy=self.taxonomy,
+            source_assembly=source_assembly
+        )
+        with open(output_file_path, 'w') as open_file:
+            open_file.write(properties)
+        return output_file_path
+
+    def create_ingestion_properties(self, output_file_path, source_assembly):
+        properties = self.properties_generator.get_remapping_ingestion_properties(
+            source_assembly=source_assembly,
+            target_assembly=self.target_assembly
+        )
+        with open(output_file_path, 'w') as open_file:
+            open_file.write(properties)
+        return output_file_path
+
+    def create_clustering_properties(self, output_file_path, instance, source_assembly):
+        properties = self.properties_generator.get_clustering_properties(
+            instance=instance,
+            source_assembly=source_assembly,
+            target_assembly=self.target_assembly,
+            rs_report_path=f'{source_assembly}_to_{self.target_assembly}_rs_report.txt'
+        )
+        with open(output_file_path, 'w') as open_file:
+            open_file.write(properties)
+        return output_file_path
+
+    def set_status(self, source_assembly, status):
+        query = (
+            f"UPDATE {self.tracking_table} "
+            f"SET remapping_status='{status}', remapping_start = '{datetime.now().isoformat()}' "
```

Comment:
You're absolutely right, these status methods are not updating the timestamps correctly...  I don't think we need an additional field as there is already `remapping_report_time` (which I believe should be set to when the tracking row is initially filled out), `remapping_start` and `remapping_end`, but we could consider adding one if it would be useful.